### PR TITLE
Team sidebar in products

### PR DESCRIPTION
**Pull request by harshilsharma63**
#### Summary
Adds the ability to show team sidebar in products/

#### Ticket Link
https://community-daily.mattermost.com/boards/workspace/qgsck6cts3fwpqwyjiupjm5cde/b4ctmm917q3rzdfi451qtuxjzcw/392c8889-6c8f-4cb7-8407-f5c57c1acfe8/680fc318-855a-4ad7-8471-fec5a79003e8

#### Related Pull Requests
A demo-only PR for interacting with the team sidebar from Boards - https://github.com/mattermost/focalboard/pull/1666/files

#### Screenshots
**Channels-** no change
Screenshot 2021-10-26 at 11 41 14 AM

**Playbooks-** no change as the product doesn't request team sidebar
Screenshot 2021-10-26 at 11 41 36 AM

**Boards-**
Screenshot 2021-10-26 at 11 42 47 AM

#### Release Note
```release-note
Added new flag showTeamSidebar in registerProducts, which, when set to true, displays the team sidebar in product.
```

### CODE REVIEW

File: plugins/registry.js
Comment (crspeller) [reply plugins/registry.js:722]: Ya that makes more sense. Then for playbooks we can verify before enabling.

Comment (harshilsharma63) [reply plugins/registry.js:722]: Exactly.

**Comment by crspeller** [comment]
@harshilsharma63 Not sure about the type error but my guess is it has something to do with connect there. You could convert the component to functional with hooks to get around it 😄

**Review by crspeller — APPROVED** [review]

File: plugins/registry.js
Comment (harshilsharma63) [reply plugins/registry.js]: Done.

File: components/team_sidebar/index.ts
Comment (harshilsharma63) [reply components/team_sidebar/index.ts:53]: @hmhealey I've undone this change to solve the "too complex type" error.

**Comment by harshilsharma63** [comment]
@hmhealey should I consider this as good to merge?

**Comment by hmhealey** [comment]
The code looks good from my perspective, so once it's got a release note (or not), it's good to merge unless there's something for QA to review now.

Regarding the duplicated logic between hooks and not, that's only natural when trying to make new tech work with older code.

**Comment by harshilsharma63** [comment]
> The code looks good from my perspective, so once it's got a release note (or not), it's good to merge unless there's something for QA to review now.
> 
> Regarding the duplicated logic between hooks and not, that's only natural when trying to make new tech work with older code.

Who can I get this QAed by?

**Comment by hmhealey** [comment]
I think it depends on who your team's QA is. I always assign mine to Jelena, but that's Web Platform

**Comment by harshilsharma63** [comment]
@ogi-m can you please test this?

This is the scope -
1. Channels should continue working as-is w.r.t. team sidebar.
2. Boards and Playbooks should NOT display team sidebar and continue working as-is.
3. This special [build of Boards](https://community-daily.mattermost.com/files/48xqz1dt3pyzxpnoif4q3636gh/public?h=pMdhyRKddnpvwhpL4JDyi5J6V9YIurCOIhxEHN_Ra9M) should display the team sidebar (it won't e used for anything though) and the Boards should continue working as-is.

File: selectors/products.ts
Comment (saturninoabril) [inline selectors/products.ts:11]: @harshilsharma63 See E2E tests [here](https://mattermost-cypress-report.s3.amazonaws.com/89397-5f06424-ee-ent-webapp-pr-9262-mm-ee-test:5f06424/mochawesome.html) and check if anything is related to this PR. One thing I noticed in the console log is likely thrown from this code block. See screenshot.
Screen Shot 2021-11-23 at 9 29 54 AM

Comment (saturninoabril) [reply selectors/products.ts:11]: This is now resolved. Thanks @harshilsharma63 for fixing!

**Comment by saturninoabril** [comment]
@harshilsharma63 See [new E2E test result](https://mattermost-cypress-report.s3.amazonaws.com/89984-dbf62f4-ee-ent-webapp-pr-9262-mm-ee-test:dbf62f4/mochawesome.html). Related failed tests are those in "back stage" pages like Integrations, Incoming/Outgoing webhooks, Slash command and Custom Emoji.

**Comment by harshilsharma63** [comment]
@saturninoabril @ogi-m The tests look good now - https://automation-dashboard.vercel.app/cycles/4ecb42e5-c8c2-4de6-8369-047abab6c004 . Can you take a look?

**Review by ogi-m — APPROVED** [review]
Thanks @harshilsharma63 , LGTM 👍🏻

**Review by saturninoabril — APPROVED** [review]
Thank you @harshilsharma63! LGTM.

**Comment by saturninoabril** [comment]
1. Text not visible in product menu in backstage page.
Screen Shot 2021-12-02 at 10 05 57 AM

**Comment by saturninoabril** [comment]
2. Theme not applied in backstage, like in modals (Settings, Profile...)
Screen Shot 2021-12-02 at 10 08 46 AM

**Comment by saturninoabril** [comment]
3. Search bar is available in backstage but doing search does not do anything.
Screen Shot 2021-12-02 at 10 09 31 AM

**Comment by saturninoabril** [comment]
4. Clicking on buttons such as at-mention and saved posts don't do anything in backstage.
Screen Shot 2021-12-02 at 10 09 54 AM

**Review by saturninoabril — CHANGES_REQUESTED** [review]
@harshilsharma63 I did check and found some UI discrepancies. Please take a look. Thanks!

**Comment by saturninoabril** [comment]
@harshilsharma63 or @ogi-m I can't access the ticket link in the description above. I wonder if we need a Jira ticket as well since this is introducing change to the core webapp. Also, I'm curious if UI/UX needs review/approval here.

**Comment by ogi-m** [comment]
You're probably right @saturninoabril, I've created a ticket for it - https://mattermost.atlassian.net/browse/MM-40464

**Comment by harshilsharma63** [comment]
Thanks, @saturninoabril , I'm taking a look.

**Comment by harshilsharma63** [comment]
@saturninoabril @ogi-m  the theme and font colour issues you mentioned are occurring because previously the global header is not intended to be shown on backstage pages. It's hidden on `master`. It shows up unintentionally in this branch. I'll hide it and that should fix the issues.

**Review by saturninoabril — APPROVED** [review]
Thanks @harshilsharma63! Tested and all looks good to me.

**Pull request merged**
